subgraphs/enzyme-core/mappings/v4/ExternalPositionManager.ts

```diff
@@ -1763,6 +1774,158 @@ export function handleCallOnExternalPositionExecutedForFund(event: CallOnExterna
     return;
   }
 
+  if (type.label == 'STAKEWISE_V3') {
+    if (actionId == StakeWiseV3StakingPositionActionId.Stake) {
+      let decoded = ethereum.decode('(address,uint256)', event.params.actionArgs);
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let stakeWiseVault = tuple[0].toAddress();
+      let amount = toBigDecimal(tuple[1].toBigInt(), 18);
+
+      let stakeWiseV3EthVault = ExternalSdk.bind(stakeWiseVault);
+      let shares = toBigDecimal(stakeWiseV3EthVault.convertToShares(tuple[1].toBigInt()), 18);
+
+      let stakeWiseVaultToken = ensureStakeWiseVaultToken(stakeWiseVault, event.params.externalPosition);
+
+      let wethAsset = ensureAsset(wethTokenAddress);
+
+      let assetAmount = createAssetAmount(wethAsset, amount, denominationAsset, 'stakewise-stake', event);
+
+      createStakeWiseStakingPositionChange(
+        event.params.externalPosition,
+        'Stake',
+        stakeWiseVaultToken,
+        assetAmount,
+        shares,
+        vault,
+        event,
+      );
+
+      let stakingPosition = useStakeWiseStakingPosition(event.params.externalPosition.toHex());
+      stakingPosition.stakedEthAmount = stakingPosition.stakedEthAmount.plus(amount);
```

Comment:
I disagree here: a lot of modifications happen in the mappings (search for `.save()` in the entities folder), and it would not be practical to write `update` functions for all entities. We typically have `create`, `ensure` and `use` functions for the entities.